pkg/alexa/l10n/l10n.go

```diff
@@ -0,0 +1,246 @@
+package l10n
+
+import (
+	"fmt"
+	"github.com/drpsychick/alexa-go-cloudformation-demo/pkg/alexa"
+	"math/rand"
+	"time"
+)
+
+// Default keys
+const (
+	KeySkillName                Key = "SKILL_Name"
+	KeySkillDescription         Key = "SKILL_Description"
+	KeySkillSummary             Key = "SKILL_Summary"
+	KeySkillExamplePhrases      Key = "SKILL_ExamplePhrases"
+	KeySkillKeywords            Key = "SKILL_Keywords"
+	KeySkillSmallIconURI        Key = "SKILL_SmallIconURI"
+	KeySkillLargeIconURI        Key = "SKILL_LargeIconURI"
+	KeySkillTestingInstructions Key = "SKILL_TestingInstructions"
+	KeySkillInvocation          Key = "SKILL_Invocation"
+	KeySkillPrivacyPolicyURL    Key = "SKILL_PrivacyPolicyURL"
+	KeySkillTermsOfUse          Key = "SKILL_TermsOfUse"
+)
+
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+func Speak(text string) string {
+	return "<speak>" + text + "</speak>"
+}
+func UseVoice(voice string, text string) string {
+	return "<voice name=\"" + voice + "\">" + text + "</voice>"
+}
+
+func UseVoiceLang(voice string, language string, text string) string {
+	return "<voice name=\"" + voice + "\"><lang xml:lang=\"" + language + "\">" + text + "</lang></voice>"
+}
+
+// Registry is the Locale registry
+type Registry struct {
+	defaultLocale string
+	locales       map[string]*Locale
+}
+
+// DefaultRegistry is the standard registry used
+var DefaultRegistry = &Registry{
+	locales: map[string]*Locale{},
+}
+
+// Locale is a representation of Keys in a specific language (and can have a fallback Locale)
+type Locale struct {
+	Name            string          // de-DE, en-US, ...
+	Countries       []alexa.Country // countries associated with this locale
+	Invocation      string          // "my skill"
+	Fallback        *Locale         // points to fallback (or nil)
+	TextSnippets    Snippets
+	IntentResponses IntentResponses
+}
+
+// Key defines the type of a text key
+type Key string
+
+func (k Key) String() string {
+	return string(k)
+}
+
+// Snippets is the actual representation of key -> array of texts in locale
+type Snippets map[Key][]string
+
+// Responses is the representation of a list of IntentResponses
+type IntentResponses map[Key]IntentResponse
+
+type IntentResponse struct {
+	Samples []string
+	Title   []string
+	Text    []string
+	SSML    []string
+	Slots   map[Key]Slot
+}
+
+type Slot struct {
```

Comment:
Do translations care about slots. Seems misplaced.

Reply:
Slots have locale specific "elicitation" prompts and samples, so yes its required for l10n.

Reply:
I would disagree. The l10n is not concerned with what it is translating, just that translations exist for keys. You are forcing skill information into a place that should have no knowledge of it.

Reply:
Yes, with the "new" (initial) approach of using simple key -> []string localization this will no longer be needed (in fact `IntentResponse` will go away.